base/twiceprecision.jl

```diff
@@ -536,6 +536,21 @@ function __convertSRL(::Type{StepRangeLen{T,R,S}}, r::AbstractRange{U}) where {T
 end
 
 function sum(r::StepRangeLen)
+    l = length(r)
+    # Compute the contribution of step over all indices.
+    # Indexes on opposite side of r.offset contribute with opposite sign,
+    #    r.step * (sum(1:np) - sum(1:nn))
+    np, nn = l - r.offset, r.offset - 1  # positive, negative
+    # To prevent overflow in sum(1:n), multiply its factors by the step
+    sp, sn = sumpair(np), sumpair(nn)
+    W = widen(Int)
+    Δn = Int(W(sp[1]) * W(sp[2]) - W(sn[1]) * W(sn[2]))
```

Comment:
You know what, I think I screwed up here. We should probably eliminate the `Int` from this, I was mentally thinking there'd have to be a cancelation here for any usable range but that's not guaranteed.

Reply:
Do you mean we don't need the widening, and we want this to simply be:
```julia
sp, sn = sumpair(np), sumpair(nn)
Δn = sp[1] * sp[2] - sn[1] * sn[2]
```
Or that we shouldn't convert back to an Int, leaving `typeof(Δn) == wident(Int)`?

Reply:
The second (don't convert back to `Int`). Since we have a `convert` in the final step, we might as well wait until then. (I was thinking, "oh, any big ranges that might overflow will also have to have cancelation on either side of the offset," but that's just wrong: multiplying by the step gives an additional mechanism for avoiding overflow.)

Reply:
Okay, I removed the `Int`. For what it's worth, because `::Float64 * ::Int128` yields `Float64`, both versions will be equivalent in the fairly common case where the step is a float (or a wrapper, like the Furlong).